middleware/domain/source/discovery/handle.cpp

```diff
@@ -497,6 +485,8 @@ namespace casual
                      return;
                   }
 
+                  state.in_flight_cache.add( message);
```

Comment:
In case of a local lookup path in below statements, I think there is a obsolete cache registration never removed.

Reply:
Yeah, I think you're right.